homeassistant/components/vallox/fan.py

```diff
@@ -61,21 +62,17 @@ def _convert_fan_speed_value(value: StateType) -> int | None:
     return None
 
 
-async def async_setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the fan device."""
-    if discovery_info is None:
-        return
-
-    client = hass.data[DOMAIN]["client"]
+    client = hass.data[DOMAIN][entry.entry_id]["client"]
     client.set_settable_address(METRIC_KEY_MODE, int)
 
     device = ValloxFan(
-        hass.data[DOMAIN]["name"], client, hass.data[DOMAIN]["coordinator"]
+        hass.data[DOMAIN][entry.entry_id]["name"],
```

Comment:
Slightly shorter if you do

data = `hass.data[DOMAIN][entry.entry_id]` above and then replace this as `data["name"]`

Reply:
Good idea, fixed.